File: Leetcode/algorithms/Java/longest palindromic substring/longest_palindromic_substring.java
class Solution {
    public String longestPalindrome(String s) {
        
        if(s.length()<2)
            return s;

        String lps="";
        for(int i=1; i< s.length(); i++){
            int left = i, right=i;

            while(s.charAt(left) == s.charAt(right)){
                left--;
                right++;
                if(left == -1 || right == s.length())
                    break;
            }

            String subStr = s.substring(left+1,right);
            if(subStr.length() > lps.length())
                lps=subStr;
            
            //for even number 
            left = i-1;
            right=i;
            while(s.charAt(left) == s.charAt(right)){
                left--;
                right++;
                if(left == -1 || right == s.length())
                    break;
            }

            subStr = s.substring(left+1,right);
            if(subStr.length() > lps.length())
                lps=subStr;

        }

        return lps;
    }
}

File: Leetcode/algorithms/JS/longest palindromic substring/longest_palindromic_substring.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    
    if(s?.length<2)
        return s;
    let longestSubString="";

    for(let i=1; i<s?.length; i++){
        let left=i,right=i;

        while(s?.[left] == s?.[right]){
            left--;
            right++;
            if(left==-1 || right>s.length-1)
                break;
        }
        console.log(left,right)
        let subString= s?.substring(left+1,right);
        if(subString?.length>longestSubString?.length)
            longestSubString=subString;

        // check for even
        left=i-1,right=i;

        while(s?.[left] == s?.[right]){
            left--;
            right++;
            if(left==-1 || right>s.length-1)
                break;
        }
        subString= s?.substring(left+1,right);
        if(subString?.length>longestSubString?.length)
            longestSubString=subString;
    }

    return longestSubString;
};